addons/mail/models/mail_mail.py

```diff
@@ -495,18 +517,23 @@ def send(self, auto_commit=False, raise_exception=False):
                     batch.write({'state': 'exception', 'failure_reason': exc})
                     batch._postprocess_sent_message(success_pids=[], failure_type="mail_smtp")
             else:
+                max_email_size = self.env['ir.mail_server'].browse(mail_server_id).max_email_size if mail_server_id \
```

Comment:
I would move this code in "prepare_outgoing_list" if we propagate the mail_server, that way we hide the low level details in that preparation method, do you think this would be ok ?